src/test/java/com/checkmarx/jira/JiraTestUtils.java

```diff
@@ -0,0 +1,337 @@
+    public Map<Filter.Severity, Integer> getIssuesPerSeverity(String projectKey) {
+        Map<Filter.Severity, Integer> result= new HashMap<>();
+        SearchResult searchResults = search(String.format("project = \"%s\"", projectKey));
+        for (Issue issue: searchResults.getIssues()) {
+            String severity = getIssueSeverity(issue.getDescription()).toUpperCase();
+            Filter.Severity filterSeverity = Filter.Severity.valueOf(severity.toUpperCase());
+            if (severity != null) {
+                if (result.containsKey(filterSeverity)) {
+                    result.put(filterSeverity,result.get(filterSeverity) + 1 );
+                } else {
+                    result.put(filterSeverity, 1);
+                }
+            }
+        }
+        return result;
+    }
+
+    private String getIssueSeverity(String issueDescription) {
+        return getIssueBodyPart(issueDescription,"Severity:");
+    }
+
+    private String getIssueBodyPart(String issueDescription, String field) {
+        String[] lines = issueDescription.split(System.lineSeparator());
+        for (String line: lines) {
+            if (line.contains(field)) {
+                return line.split(" ")[1];
+            }
+        }
+        return null;
+    }
+
+    @Override
+    public String getIssueFilename(String projectKey) {
+        Issue issue = getFirstIssue(projectKey);
+        String firstLine = issue.getDescription().split(System.lineSeparator())[0];
+        String[] firstLineParts= firstLine.split(" ");
+        return firstLineParts[4];
+    }
+
+    // TODO UDI: check if this is realy the vulnerability field !
+    @Override
+    public String getIssueVulnerability(String projectKey) {
+        Issue issue = getFirstIssue(projectKey);
+        String firstLine = issue.getDescription().split(System.lineSeparator())[0];
+        String[] firstLineParts= firstLine.split(" ");
+        return firstLineParts[0];
+    }
+
+
+    @Override
+    public int getFirstIssueNumOfFindings(String projectKey) {
+        SearchResult result = search(String.format("project = \"%s\"", projectKey));
+        if (result.getTotal() ==0) {
+            return 0;
+        }
+        Issue i = result.getIssues().iterator().next();
+        int lastIndex = 0;
+        int count = 0;
+        while (lastIndex != -1) {
+            lastIndex = i.getDescription().indexOf(JIRA_DESCRIPTION_FINDING_LINE, lastIndex);
+            if (lastIndex != -1) {
+                count++;
+                lastIndex += JIRA_DESCRIPTION_FINDING_LINE.length();
+            }
+        }
+        return count;
+    }
+    @Override
+    public void ensureProjectExists(String key) throws IOException {
+        log.info("Making sure '{}' project exists in Jira.", key);
+        ResourceCreationConfig config = getProjectCreationConfig(key);
+        tryCreateResource(config);
+    }
+
+    @Override
+    public void ensureIssueTypeExists(String issueType) throws IOException {
```

Comment:
Is ensureIssueTypeExists still used?

Reply:
Well, I made the call to this method a comment, because there were some problems with this method (different JIRA's, String parsing if you remember).
I think we do need this method, but I thought you had fixed it.
Anyway, I leave it here for now and we can discuss it later on.
I really want to merge the branch to develop today.